workflow/graph/topsort.go

```diff
@@ -0,0 +1,57 @@
+package graph
+
+import (
+	"github.com/stevenle/topsort"
+
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+)
+
+// Return a list of node IDs that can be iterated over know we'll visit each one only once and that
+// we'll visit downstream dependencies before upstream ones.
+// Children can be:
+// * "missing" - appear in `node.Children` but missing from `nodes` - callers must cope with this and probably skip them
+func TopSort(nodes wfv1.Nodes) ([]string, error) {
+	g, err := graph(nodes)
```

Comment:
@simster7 pursuant to our chat, this shows the API I'd like. I'm not really fussed about the implementation (as long as it works).